cpp/daal/src/services/compiler/generic/env_detect_features.cpp

```diff
@@ -71,6 +71,19 @@ void run_cpuid(uint32_t eax, uint32_t ecx, uint32_t * abcd)
     #endif
 }
 
+uint32_t __daal_internal_get_max_extension_support()
```

Comment:
This function is not necessary, it can be folded in below.

Reply:
This follows the pattern of `__daal_internal_is_intel_cpu` and `daal_check_is_intel_cpu` where the former is the one that does the work and the latter is there to cache the value in a static variable to avoid running the former multiple times.